src/http/httpd.cc

```diff
@@ -473,14 +473,31 @@ void http_server::set_content_streaming(bool b) {
     _content_streaming = b;
 }
 
-future<> http_server::listen(socket_address addr, listen_options lo) {
-    if (_credentials) {
-        _listeners.push_back(seastar::tls::listen(_credentials, addr, lo));
+future<> http_server::listen(socket_address addr, listen_options lo, 
+            shared_ptr<seastar::tls::server_credentials> listener_credentials) {
+    if(listener_credentials) {
```

Comment:
I wonder if the semantics should be that a null `listener_credentials` disables the default `_credentials`?

Reply:
good point, we should probably discuss this, it would require a little bit more code